pkg/cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -121,8 +121,12 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 	}
 
 	defer client.Close()
-
-	return deleteVolumeAndDomain(machine, client)
+	exists, err := libvirtutils.DomainExists(machine.Name, client)
+	if exists {
+		return deleteVolumeAndDomain(machine, client)
+	}
+	glog.Infof("Machine does not exist. Skipping deletion...")
```

Comment:
Given the context of the actuator it could make more sense to use `domain` instead of `machine`. E.g.
```go
glog.Infof("Domain %v does not exist. Skipping deletion...", machine.Name)
```

WDYT?

Reply:
sg

Reply:
this seems still counterintuitive syntax to me, before checking `if exists` we should check the error
```
	exists, err := libvirtutils.DomainExists(machine.Name, client)
	if err != nil {
		return err
	}
	if exists {
		return deleteVolumeAndDomain(machine, client)
	}
	glog.Infof("Machine does not exist. Skipping deletion...")
	return nil
```